api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -564,72 +560,59 @@ public RelationshipType saveRelationshipType(RelationshipType relationshipType)
 	 * @see org.openmrs.api.PersonService#getPersonAttributeTypes(org.openmrs.util.OpenmrsConstants.PERSON_TYPE,
 	 *      org.openmrs.api.PersonService.ATTR_VIEW_TYPE)
 	 */
+	
+	private String userPatientAttributes(String patientAttributeProperty, String userAttributeProperty,
+	        PERSON_TYPE personType) {
+		StringBuilder property = new StringBuilder();
+		
+		if (personType == null || personType == PERSON_TYPE.PERSON) {
+			property.append(patientAttributeProperty).append(",").append(userAttributeProperty);
+		} else if (personType == PERSON_TYPE.PATIENT) {
+			property.append(patientAttributeProperty);
+		} else if (personType == PERSON_TYPE.USER) {
+			property.append(userAttributeProperty);
+		} else {
+			log.error(MarkerFactory.getMarker("FATAL"), "Should not be here.");
+		}
+		return property.toString();
+	}
+	
 	@Override
 	@Transactional(readOnly = true)
 	public List<PersonAttributeType> getPersonAttributeTypes(PERSON_TYPE personType, ATTR_VIEW_TYPE viewType)
 	        throws APIException {
 		AdministrationService as = Context.getAdministrationService();
 		
 		String attrString = "";
-		
-		final String fatalString = "Should not be here.";
-		
+    
+    final String fatalString = "Should not be here.";		
+
 		// TODO cache the global properties to speed this up??
 		// Is hibernate taking care of caching and not hitting the db every time? (hopefully it is)
 		if (viewType == null) {
 			return Context.getPersonService().getAllPersonAttributeTypes();
 		} else if (viewType == ATTR_VIEW_TYPE.LISTING) {
 			String patientListing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_PATIENT_LISTING_ATTRIBUTES, "");
 			String userListing = as.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_USER_LISTING_ATTRIBUTES, "");
-			if (personType == null || personType == PERSON_TYPE.PERSON) {
-				attrString = patientListing + "," + userListing;
-			} else if (personType == PERSON_TYPE.PATIENT) {
-				attrString = patientListing;
-			} else if (personType == PERSON_TYPE.USER) {
-				attrString = userListing;
-			} else {
-				log.error(MarkerFactory.getMarker("FATAL"), fatalString);
-			}
+      attrString = userPatientAttributes(userListing, patientListing, personType);
```

Comment:
please configure your IDE according to https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE
your formatting is off